cli/tests/test_observability_v8_writer.py

```
@@ -207,6 +207,110 @@ def validator(_candidate: str, _data_dir: str | None) -> None:
+
+    assert stat.S_IMODE(backup_parent.stat().st_mode) == 0o755
+
+
+@pytest.mark.skipif(os.name == "nt", reason="hard-link semantics differ on Windows")
+def test_mutate_v8_config_rejects_backup_hard_linked_to_source(tmp_path: Path) -> None:
+    path = tmp_path / "config.yaml"
+    path.write_text(_source())
+    backup = tmp_path / "config.yaml.before-redaction"
+    os.link(path, backup)
+
+    with pytest.raises(ValueError, match="must not alias config.yaml"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)